addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/command/LiveDataUpdatePublicApi.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.solaredge.internal.command;
+
+import static org.openhab.binding.solaredge.SolarEdgeBindingConstants.*;
+
+import java.nio.charset.StandardCharsets;
+
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Result;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.solaredge.handler.SolarEdgeHandler;
+import org.openhab.binding.solaredge.internal.callback.AbstractCommandCallback;
+import org.openhab.binding.solaredge.internal.model.LiveDataResponse;
+
+/**
+ * command that retrieves status values for live data channels via public API
+ *
+ * @author Alexander Friese - initial contribution
+ */
+public class LiveDataUpdatePublicApi extends AbstractCommandCallback implements SolarEdgeCommand {
+
+    private final SolarEdgeHandler handler;
+    private int retries = 0;
+
+    public LiveDataUpdatePublicApi(SolarEdgeHandler handler) {
+        super(handler.getConfiguration());
+        this.handler = handler;
+    }
+
+    @Override
+    protected Request prepareRequest(Request requestToPrepare) {
+        requestToPrepare.followRedirects(false);
+        requestToPrepare.method(HttpMethod.GET);
+
+        return requestToPrepare;
+    }
+
+    @Override
+    protected String getURL() {
+        return PUBLIC_DATA_API_URL + config.getSolarId() + PUBLIC_DATA_API_URL_LIVE_DATA_SUFFIX;
+    }
+
+    @Override
+    public void onComplete(Result result) {
+        logger.debug("onComplete()");
+
+        if (!HttpStatus.Code.OK.equals(getCommunicationStatus().getHttpCode())) {
+            if (getListener() != null) {
+                getListener().update(getCommunicationStatus());
+            }
+            if (retries++ < MAX_RETRIES) {
+                handler.getWebInterface().executeCommand(this);
+            }
+
```

Comment:
you might want to remove some empty lines here...

Reply:
done